polling_stations/apps/uk_geo_utils/models.py

```diff
@@ -98,3 +98,114 @@ class Meta:
+    psed = models.CharField(blank=True, max_length=8)
+    cened = models.CharField(blank=True, max_length=6)
+    edind = models.CharField(blank=True, max_length=1)
+    oshaprev = models.CharField(blank=True, max_length=3)
+    lea = models.CharField(blank=True, max_length=3)
+    oldha = models.CharField(blank=True, max_length=3)
+    wardc91 = models.CharField(blank=True, max_length=6)
+    wardo91 = models.CharField(blank=True, max_length=6)
+    ward98 = models.CharField(blank=True, max_length=6)
+    statsward = models.CharField(blank=True, max_length=6)
+    oa01 = models.CharField(blank=True, max_length=10)
+    casward = models.CharField(blank=True, max_length=6)
+    park = models.CharField(blank=True, max_length=9)
+    lsoa01 = models.CharField(blank=True, max_length=9)
+    msoa01 = models.CharField(blank=True, max_length=9)
+    ur01ind = models.CharField(blank=True, max_length=1)
+    oac01 = models.CharField(blank=True, max_length=3)
+    oldpct = models.CharField(blank=True, max_length=5)
+    oa11 = models.CharField(blank=True, max_length=9)
+    lsoa11 = models.CharField(blank=True, max_length=9)
+    msoa11 = models.CharField(blank=True, max_length=9)
+    parish = models.CharField(blank=True, max_length=9)
+    wz11 = models.CharField(blank=True, max_length=9)
+    ccg = models.CharField(blank=True, max_length=9)
+    bua11 = models.CharField(blank=True, max_length=9)
+    buasd11 = models.CharField(blank=True, max_length=9)
+    ru11ind = models.CharField(blank=True, max_length=2)
+    oac11 = models.CharField(blank=True, max_length=3)
+    lat = models.CharField(blank=True, max_length=10)
+    long = models.CharField(blank=True, max_length=10)
+    lep1 = models.CharField(blank=True, max_length=9)
+    lep2 = models.CharField(blank=True, max_length=9)
+    pfa = models.CharField(blank=True, max_length=9)
+    imd = models.CharField(blank=True, max_length=5)
+    location = models.PointField(null=True, blank=True)
+    objects = models.GeoManager()
+
+    def _get_cty(self):
+        """
+        Note: these are not _exactly_ the same because
+
+        'CTY' in ONSUD is:
+        E10000002 - E10000034 = England (county);
+        E11000001 - E11000007 = England (metropolitan county);
+        E13000001 - E13000002 = England (Inner/Outer London);
+        E99999999 (pseudo) = England (UA/MD/LB);
+        W99999999 (pseudo) = Wales;
+        S99999999 (pseudo) = Scotland
+
+        whereas 'OSCTY' in ONSPD is
+        E10000002 – E10000034 = England;
+        E99999999 (pseudo) = England (UA/MD/LB);
+        W99999999 (pseudo) = Wales;
+        S99999999 (pseudo) = Scotland;
+        """
+        return self.oscty
+
+    def _get_lad(self):
+        return self.oslaua
+
+    def _get_ward(self):
+        return self.osward
+
+    def _get_hlthau(self):
+        return self.oshlthau
+
+    def _get_rgn(self):
+        return self.gor
+
+    def _get_ruc11(self):
+        return self.ru11ind
+
+    cty = property(_get_cty)
```

Comment:
No problems here, but what's the thinking behind this pattern rather than just using `@property` on `def cty(self):`?

Reply:
Now you raise it, I think I just looked up how to alias fields on a django model and found a code example that used the function form of `property` rather than the decorator form of `property`. These could probably be rewritten using the decorator form.

Reply:
Interesting they've done it that way! I wonder if this is because there's issues with sub-classing a property?